File: All Lab Tasks/Lab 1/RideSharingApp__Strategy_Pattern_/src/PayPalPayment.java
public class PayPalPayment implements IPaymentMethod
{
    private String email;
    private int otp;

    public PayPalPayment(String email, int otp) {
        this.email = email;
        this.otp = otp;
    }
    public String getEmail() {
        return email;
    }

    public int getOtp() {
        return otp;
    }
    @Override
    public void ProcessPayment(double amount) {
        System.out.println("Processing PayPal payment of " + amount);
        System.out.println("Email: " + email + ", OTP: " + otp);
    }
}

File: Lab_1_RideSharingApp/Lab_1_RideSharingApp/Payment/PayPalPayment.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab_1_RideSharingApp
{
    public class PayPalPayment : IPaymentMethod
    {
        public string Email { get; private set; }
        public int Otp { get; private set; }

        // Constructor
        public PayPalPayment(string email, int otp)
        {
            Email = email;
            Otp = otp;
        }

        // Implementing IPaymentMethod interface
        public void ProcessPayment(double amount)
        {
            Console.WriteLine($"Processing PayPal payment of {amount}.");
            Console.WriteLine($"Email: {Email}, OTP: {Otp}");
        }
    }
}
